durable_storage/src/persistence_layer.rs

```diff
@@ -0,0 +1,93 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+//! Implementation of the RISC-V PVM durable storage's persistence layer.
+//!
+//! A persistence layer is tied to a repository on disk, identified by a directory.
+//! Within that directory, the persistence layer needs to be able to perform basic KV operations:
+//! - get
+//! - set
+//! - delete
+//!
+//! As well as repository-level operations:
+//! - new
+//! - clone
+//! - commit (returning the commit hash)
+//! - checkout a specific commit
+
+use std::path::Path;
+
+/// Type alias for a 32-byte hash used for identifying key-value blobs & commits.
+type Hash = [u8; 32];
+
+/// Represents a key-value blob to be stored in the persistence layer.
+///
+/// Invariant: A [`KVBlob`] is a content-addressable blob.
+pub struct KVBlob {
+    /// The key is the blake3 hash of the value.
+    key: Hash,
+
+    /// The addressable content of the blob.
+    value: Vec<u8>,
+}
+
+impl KVBlob {
+    pub fn from_value(value: Vec<u8>) -> Self {
+        let digest = blake3::hash(&value).into();
+        Self { key: digest, value }
+    }
+}
+
+/// Errors encountered when interacting with the persistence layer.
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("Key not found")]
+    KeyNotFound,
+}
+
+/// Persistence layer for durable solution used by the RISC-V PVM.
+pub struct PersistenceLayer {}
+
+impl PersistenceLayer {
+    /// Creates a new `PersistenceLayer` instance within the given `repo`.
+    pub fn new(_repo: &Path) -> Result<Self, Error> {
+        todo!("TODO: RV-793 - implement the persistence layer")
+    }
+
+    /// Clones the current `PersistenceLayer` instance.
+    ///
+    /// Operations on the cloned instance will have no effect on the original instance.
+    fn try_clone(&self) -> Result<Self, Error> {
```

Comment:
What is the error that can arise when cloning?

Reply:
Checkpoint operations can still error out. Assuming folder creation / deletion is handled error-less, is it acceptable to panic if rocks still returns an error?